x-pack/plugins/security_solution/public/actions/add_to_timeline/default/add_to_timeline.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CellActionExecutionContext } from '@kbn/ui-actions-plugin/public';
+import { createAction } from '@kbn/ui-actions-plugin/public';
+import { addProvider } from '../../../timelines/store/timeline/actions';
+import { TimelineId } from '../../../../common/types';
+import { KibanaServices } from '../../../common/lib/kibana';
+import type { SecurityAppStore } from '../../../common/store';
+import { isInSecurityApp } from '../../utils';
+import {
+  ADD_TO_TIMELINE,
+  ADD_TO_TIMELINE_FAILED_TEXT,
+  ADD_TO_TIMELINE_FAILED_TITLE,
+  ADD_TO_TIMELINE_ICON,
+  ADD_TO_TIMELINE_SUCCESS_TITLE,
+} from '../constants';
+import { createDataProviders } from '../data_provider';
+
+export const ACTION_ID = 'security_addToTimeline';
+
+export const createAddToTimelineAction = ({
+  store,
+  order,
+}: {
+  store: SecurityAppStore;
+  order?: number;
+}) => {
+  const { application: applicationService, notifications: notificationsService } =
+    KibanaServices.get();
+  let currentAppId: string | undefined;
+  applicationService.currentAppId$.subscribe((appId) => {
+    currentAppId = appId;
+  });
+
+  return createAction<CellActionExecutionContext>({
```

Comment:
We did a bit of action clean-up here https://github.com/elastic/kibana/pull/148037 with a way how actions should be created. 

I believe the current proper way to not use the `createAction` helper (we should remove it) and instead create these actions as `ActionDefinition` that are registered in the action service. 


Same applies to other actions.

We can also do  this change when we get to removing `createAction` helper

Reply:
Thank you for the heads-up. I will do this change in a follow-up PR.